specgen/specgen/main.go

```diff
@@ -0,0 +1,117 @@
+// Command specgen provides a utility for generating utls.ClientHelloSpecs. This command starts an
+// HTTPS server which responds to every request with a ClientHelloSpec corresponding to the TLS
+// ClientHello the server received. The spec is logged for every request as well. Make a request of
+// the server via web browser to see a ClientHelloSpec for the browser.
+//
+// It is recommended that mkcert be used to generate the keypair for this server. This ensures
+// browsers do not flag and block the page. See https://github.com/FiloSottile/mkcert.
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/getlantern/utls/specgen"
+	"github.com/getlantern/utls/specgen/specgen/hellocap"
+)
+
+const certFileDefaultText = "pre-defined, self-signed cert"
```

Comment:
Don't think you need this. You can just check for the empty string (or nil, since you're using pointer flags).

Reply:
This is intentionally not the empty string - for the usage output.

Reply:
Ah, gotcha. I'm used to writing my own usage so I didn't catch that.